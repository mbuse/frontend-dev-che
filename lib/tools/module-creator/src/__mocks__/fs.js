"use strict";

const path = require("path");

const fs = jest.genMockFromModule("fs");

let mockFiles = Object.create(null);
fs.__setMockFiles = (newMockFiles) => {
  mockFiles = newMockFiles.reduce((files, file) => {
    const dir = path.dirname(file).replace(/\\/g, "/");
    if (!files[dir]) {
      files[dir] = [];
    }
    files[dir].push(path.basename(file));
    return files;
  }, Object.create(null));
};

fs.__resetMockFiles = () => {
  mockFiles = Object.create(null);
};

fs.__getMockFiles = () => mockFiles;

let mockDirectories;
fs.__resetMockDirectories = () => {
  mockDirectories = [];
};

fs.__getMockDirectories = () => mockDirectories;

fs.mkdirSync = (directory) => {
  const dir = directory.replace(/\\/g, "/");
  mockDirectories.push(dir);
};

fs.existsSync = (directory) => {
  const dir = path.dirname(directory);
  const name = path.basename(directory);
  return Object.keys(mockFiles).includes(dir) && mockFiles[dir].includes(name);
};

fs.readdirSync = (directory) => mockFiles[directory] || [];

fs.writeFileSync = (file, data) => {
  const dir = path.dirname(file).replace(/\\/g, "/");
  if (!mockFiles[dir]) {
    mockFiles[dir] = {};
  }
  mockFiles[dir][path.basename(file)] = data;
};

module.exports = fs;
