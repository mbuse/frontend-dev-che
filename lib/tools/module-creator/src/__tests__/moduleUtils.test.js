"use strict";

jest.mock("fs");

const log = {
  debug: jest.fn(),
  info: jest.fn(),
  warn: jest.fn(),
  error: jest.fn(),
  success: jest.fn(),
};

const wsConfig = {
  themesPath: "/path/to/themes/",
  bricksPath: "/path/to/bricks/",
};

describe("convertModuleName()", () => {
  const { convertModuleName } = require("../moduleUtils");
  it("returns converted theme name", () => {
    const themeName =
      ' 1234567890_-Ugly Theme Name! äöüß^°!"§$%&/()=?´`*+#.:,;<>@„¡“¶¢[]|{}”≠¿±‘’æœ•–…∞µ~√ç≈¥≤åº∆π¨Ω†®€∑«»';
    const expected = "1234567890-uglythemename";
    expect(convertModuleName(themeName)).toEqual(expected);
  });
  it("returns empty string", () => {
    const themeName =
      ' äöüß^°!"§$%&/()=?´`*+#.:,;<>@„¡“¶¢[]|{}”≠¿±‘’æœ•–…∞µ~√ç≈¥≤åº∆π¨Ω†®€∑«»';
    expect(convertModuleName(themeName)).toEqual("");
  });
  it("returns empty string", () => {
    const themeName = [1, 2];
    expect(convertModuleName(themeName)).toEqual("");
  });
  it("returns empty string", () => {
    const themeName = {};
    expect(convertModuleName(themeName)).toEqual("");
  });
});

describe("isModuleNameInUse()", () => {
  beforeEach(() => {
    jest.resetModules();
    require("fs").__setMockFiles([
      "/path/to/themes/aurora-theme",
      "/path/to/themes/corporate-theme",
      "/path/to/themes/hybris-theme",
    ]);
  });

  it("returns true", () => {
    const { isModuleNameInUse } = require("../moduleUtils");
    expect(isModuleNameInUse("/path/to/themes/aurora-theme")).toBe(true);
  });
  it("returns false", () => {
    const { isModuleNameInUse } = require("../moduleUtils");
    expect(isModuleNameInUse("/path/to/themes/newest-theme")).toBe(false);
  });
});

describe("createThemeFolderStructure()", () => {
  it("creates folder structure of a new theme", () => {
    const fs = require("fs");
    fs.__resetMockDirectories();

    const { createThemeFolderStructure } = require("../moduleUtils");
    createThemeFolderStructure(wsConfig, "/path/to/themes/newest-theme", log);
    const directories = fs.__getMockDirectories();
    expect(directories).toMatchSnapshot();
  });
});

describe("createBrickFolderStructure()", () => {
  it("creates folder structure of a new brick", () => {
    const fs = require("fs");
    fs.__resetMockDirectories();

    const { createBrickFolderStructure } = require("../moduleUtils");
    createBrickFolderStructure(wsConfig, "/path/to/bricks/newest-brick", log);
    const directories = fs.__getMockDirectories();
    expect(directories).toMatchSnapshot();
  });
});

describe("createThemeFiles()", () => {
  it("creates files of a new theme including bricks", () => {
    const fs = require("fs");
    fs.__resetMockFiles();

    const { createThemeFiles } = require("../moduleUtils");
    createThemeFiles(
      "/path/to/themes/newest-theme",
      "newest",
      { "active-brick": "^1.0.0" },
      { "some-brick": "^1.0.0" },
      "",
      log
    );
    const files = fs.__getMockFiles();
    expect(files).toMatchSnapshot();
  });
});

describe("createBrickFiles()", () => {
  it("creates files of a new brick", () => {
    const fs = require("fs");
    fs.__resetMockFiles();

    const { createBrickFiles } = require("../moduleUtils");
    createBrickFiles("/path/to/brick/newest-brick", "newest", log);
    const files = fs.__getMockFiles();
    expect(files).toMatchSnapshot();
  });
});

describe("createTheme()", () => {
  it("creates folder structure and files of a new empty theme", () => {
    const fs = require("fs");
    fs.__resetMockDirectories();
    fs.__resetMockFiles();

    const { createTheme } = require("../moduleUtils");
    createTheme(
      wsConfig,
      "/path/to/themes/newest-theme",
      "newest",
      {},
      {},
      "",
      log
    );

    const directories = fs.__getMockDirectories();
    const files = fs.__getMockFiles();

    expect(directories).toMatchSnapshot();
    expect(files).toMatchSnapshot();
  });
});

describe("createTheme() with parent theme", () => {
  it("creates folder structure and files of a new theme", () => {
    const fs = require("fs");
    fs.__resetMockDirectories();
    fs.__resetMockFiles();

    const { createTheme } = require("../moduleUtils");
    createTheme(
      wsConfig,
      "/path/to/themes/newest-theme",
      "newest",
      { "parent-theme": "^1.0.0" },
      {},
      "parent-theme",
      log
    );

    const directories = fs.__getMockDirectories();
    const files = fs.__getMockFiles();

    expect(directories).toMatchSnapshot();
    expect(files).toMatchSnapshot();
  });
});

describe("createBrick()", () => {
  it("creates folder structure and files of a new brick", () => {
    const fs = require("fs");
    fs.__resetMockDirectories();
    fs.__resetMockFiles();

    const { createBrick } = require("../moduleUtils");
    createBrick(wsConfig, "/path/to/brick/newest-brick", "newest", log);

    const directories = fs.__getMockDirectories();
    const files = fs.__getMockFiles();

    expect(directories).toMatchSnapshot();
    expect(files).toMatchSnapshot();
  });
});
