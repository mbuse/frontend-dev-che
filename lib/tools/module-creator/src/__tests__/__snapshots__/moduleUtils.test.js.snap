// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createBrick() creates folder structure and files of a new brick 1`] = `
Array [
  "/path/to/brick/newest-brick/src/js",
  "/path/to/brick/newest-brick/src/sass",
  "/path/to/brick/newest-brick/src/sass/partials",
  "/path/to/brick/newest-brick/src/sass/variables",
  "/path/to/brick/newest-brick/src/img",
  "/path/to/brick/newest-brick/src/l10n",
  "/path/to/brick/newest-brick/src/templates/com.coremedia.blueprint.common.contentbeans",
]
`;

exports[`createBrick() creates folder structure and files of a new brick 2`] = `
Object {
  "/path/to/brick/newest-brick": Object {
    "package.json": "{
  \\"name\\": \\"@coremedia/brick-newest\\",
  \\"version\\": \\"1.0.0\\",
  \\"private\\": true,
  \\"scripts\\": {
    \\"prettier\\": \\"prettier --write \\\\\\"src/**/*.{js,json}\\\\\\"\\"
  },
  \\"devDependencies\\": {
    \\"prettier\\": \\"^2.4.1\\"
  },
  \\"__comment__dependencies__\\": {
    \\"__comment__\\": \\"List of dependencies for the commented out example code. In order to add a dependency, move the entry to 'dependencies'\\",
    \\"@coremedia/brick-utils\\": \\"workspace:*\\",
    \\"jquery\\": \\"^3.6.0\\"
  },
  \\"dependencies\\": {},
  \\"main\\": \\"src/js/index.js\\",
  \\"coremedia\\": {
    \\"type\\": \\"brick\\",
    \\"init\\": \\"src/js/init.js\\"
  }
}",
  },
  "/path/to/brick/newest-brick/src/js": Object {
    "index.js": "import \\"./newest.js\\";
",
    "init.js": "//import $ from \\"jquery\\";
import { consolePrint } from \\"./newest\\";
// --- JQUERY DOCUMENT READY -------------------------------------------------------------------------------------------
//$(function () {
//  consolePrint(\\"Brick newest is used.\\");
//});
",
    "newest.js": "//import * as utils from \\"@coremedia/brick-utils\\";

/**
 * Displays a simple text in the console.
 *
 * @function consolePrint
 * @param {String} $text - The text to be displayed in the console.
 */
export function consolePrint($text) {
//  utils.log($text);
}
",
  },
  "/path/to/brick/newest-brick/src/l10n": Object {
    "Newest_de.properties": "welcomeText=Hallo Welt!",
    "Newest_en.properties": "welcomeText=Hello World!",
  },
  "/path/to/brick/newest-brick/src/sass": Object {
    "_partials.scss": "// make sure to import partials sass files in _partials.scss
// the smart-import ensures, that all sass partials from depending bricks are loaded
@import \\"?smart-import-partials\\";
@import \\"partials/newest\\";

",
    "_variables.scss": "// make sure to import variables sass files in _variables.scss
@import \\"variables/newest\\";
// the smart-import ensures, that all sass variables from depending bricks are loaded
@import \\"?smart-import-variables\\";
",
  },
  "/path/to/brick/newest-brick/src/sass/partials": Object {
    "_newest.scss": "// css rules in partials may use variables, defined in the sass/variables folder
.custom-text {
  color: $custom-text-color;
}
",
  },
  "/path/to/brick/newest-brick/src/sass/variables": Object {
    "_newest.scss": "// brick scss variables to be used in partials files
// use the !default flag to make this variable configurable in themes
$custom-text-color: #FF0000 !default;
",
  },
  "/path/to/brick/newest-brick/src/templates/com.coremedia.blueprint.common.contentbeans": Object {
    "Page._body.ftl": "<#-- Use cm.getMessage to display a localized hello world message -->
<div>
  <span class=\\"custom-text\\">\${cm.getMessage('welcomeText')}</span>
</div>
",
  },
}
`;

exports[`createBrickFiles() creates files of a new brick 1`] = `
Object {
  "/path/to/brick/newest-brick": Object {
    "package.json": "{
  \\"name\\": \\"@coremedia/brick-newest\\",
  \\"version\\": \\"1.0.0\\",
  \\"private\\": true,
  \\"scripts\\": {
    \\"prettier\\": \\"prettier --write \\\\\\"src/**/*.{js,json}\\\\\\"\\"
  },
  \\"devDependencies\\": {
    \\"prettier\\": \\"^2.4.1\\"
  },
  \\"__comment__dependencies__\\": {
    \\"__comment__\\": \\"List of dependencies for the commented out example code. In order to add a dependency, move the entry to 'dependencies'\\",
    \\"@coremedia/brick-utils\\": \\"workspace:*\\",
    \\"jquery\\": \\"^3.6.0\\"
  },
  \\"dependencies\\": {},
  \\"main\\": \\"src/js/index.js\\",
  \\"coremedia\\": {
    \\"type\\": \\"brick\\",
    \\"init\\": \\"src/js/init.js\\"
  }
}",
  },
  "/path/to/brick/newest-brick/src/js": Object {
    "index.js": "import \\"./newest.js\\";
",
    "init.js": "//import $ from \\"jquery\\";
import { consolePrint } from \\"./newest\\";
// --- JQUERY DOCUMENT READY -------------------------------------------------------------------------------------------
//$(function () {
//  consolePrint(\\"Brick newest is used.\\");
//});
",
    "newest.js": "//import * as utils from \\"@coremedia/brick-utils\\";

/**
 * Displays a simple text in the console.
 *
 * @function consolePrint
 * @param {String} $text - The text to be displayed in the console.
 */
export function consolePrint($text) {
//  utils.log($text);
}
",
  },
  "/path/to/brick/newest-brick/src/l10n": Object {
    "Newest_de.properties": "welcomeText=Hallo Welt!",
    "Newest_en.properties": "welcomeText=Hello World!",
  },
  "/path/to/brick/newest-brick/src/sass": Object {
    "_partials.scss": "// make sure to import partials sass files in _partials.scss
// the smart-import ensures, that all sass partials from depending bricks are loaded
@import \\"?smart-import-partials\\";
@import \\"partials/newest\\";

",
    "_variables.scss": "// make sure to import variables sass files in _variables.scss
@import \\"variables/newest\\";
// the smart-import ensures, that all sass variables from depending bricks are loaded
@import \\"?smart-import-variables\\";
",
  },
  "/path/to/brick/newest-brick/src/sass/partials": Object {
    "_newest.scss": "// css rules in partials may use variables, defined in the sass/variables folder
.custom-text {
  color: $custom-text-color;
}
",
  },
  "/path/to/brick/newest-brick/src/sass/variables": Object {
    "_newest.scss": "// brick scss variables to be used in partials files
// use the !default flag to make this variable configurable in themes
$custom-text-color: #FF0000 !default;
",
  },
  "/path/to/brick/newest-brick/src/templates/com.coremedia.blueprint.common.contentbeans": Object {
    "Page._body.ftl": "<#-- Use cm.getMessage to display a localized hello world message -->
<div>
  <span class=\\"custom-text\\">\${cm.getMessage('welcomeText')}</span>
</div>
",
  },
}
`;

exports[`createBrickFolderStructure() creates folder structure of a new brick 1`] = `
Array [
  "/path/to/bricks/newest-brick/src/js",
  "/path/to/bricks/newest-brick/src/sass",
  "/path/to/bricks/newest-brick/src/sass/partials",
  "/path/to/bricks/newest-brick/src/sass/variables",
  "/path/to/bricks/newest-brick/src/img",
  "/path/to/bricks/newest-brick/src/l10n",
  "/path/to/bricks/newest-brick/src/templates/com.coremedia.blueprint.common.contentbeans",
]
`;

exports[`createTheme() creates folder structure and files of a new empty theme 1`] = `
Array [
  "/path/to/themes/newest-theme/src/js",
  "/path/to/themes/newest-theme/src/sass",
  "/path/to/themes/newest-theme/src/sass/partials",
  "/path/to/themes/newest-theme/src/sass/variables",
  "/path/to/themes/newest-theme/src/img",
  "/path/to/themes/newest-theme/src/fonts",
  "/path/to/themes/newest-theme/src/l10n",
  "/path/to/themes/newest-theme/src/settings",
  "/path/to/themes/newest-theme/src/templates/com.coremedia.blueprint.common.contentbeans",
]
`;

exports[`createTheme() creates folder structure and files of a new empty theme 2`] = `
Object {
  "/path/to/themes/newest-theme": Object {
    "package.json": "{
  \\"name\\": \\"@coremedia/newest-theme\\",
  \\"version\\": \\"1.0.0\\",
  \\"private\\": true,
  \\"scripts\\": {
    \\"build\\": \\"cross-env NODE_OPTIONS=--openssl-legacy-provider webpack\\",
    \\"deploy\\": \\"cm theme-importer deploy-theme\\",
    \\"start\\": \\"cm monitor\\",
    \\"prettier\\": \\"prettier --write \\\\\\"src/**/*.{js,json}\\\\\\"\\",
    \\"theme-importer\\": \\"cm theme-importer\\"
  },
  \\"dependencies\\": {
    \\"@coremedia/cm-cli\\": \\"workspace:2.0.0\\",
    \\"@coremedia/theme-utils\\": \\"workspace:3.0.0\\",
    \\"cross-env\\": \\"^7.0.3\\",
    \\"webpack\\": \\"^4.46.0\\"
  },
  \\"devDependencies\\": {
    \\"prettier\\": \\"^2.4.1\\"
  },
  \\"main\\": \\"src/js/index.js\\",
  \\"coremedia\\": {
    \\"type\\": \\"theme\\",
    \\"init\\": \\"src/js/newest.js\\"
  },
  \\"browserslist\\": [
    \\"last 1 Chrome version\\",
    \\"last 1 Firefox version\\",
    \\"last 1 Edge version\\"
  ]
}",
    "theme.config.json": "{
  \\"name\\": \\"newest\\",
  \\"styles\\": [
    {
      \\"type\\": \\"webpack\\",
      \\"src\\": \\"src/sass/newest.scss\\"
    }
  ],
  \\"scripts\\": [
    {
      \\"type\\": \\"webpack\\",
      \\"src\\": \\"src/js/newest.js\\"
    }
  ],
  \\"l10n\\": {
    \\"bundleEncoding\\": \\"UTF-8\\",
    \\"bundleNames\\": [
      \\"Newest\\"
    ]
  }
}",
    "webpack.config.js": "const { webpackConfig } = require(\\"@coremedia/theme-utils\\");

module.exports = webpackConfig;",
  },
  "/path/to/themes/newest-theme/src/js": Object {
    "index.js": "/*! Theme newest */
import \\"./newest.js\\";
",
    "newest.js": "//import $ from \\"jquery\\";
//import * as utils from \\"@coremedia/brick-utils\\";

// --- JQUERY DOCUMENT READY -------------------------------------------------------------------------------------------

//$(function () {
//  utils.log(\\"Theme newest is used.\\");
//});
",
    "preview.js": "/*! Theme newest: Preview JS */
// add preview specific code here...
",
  },
  "/path/to/themes/newest-theme/src/l10n": Object {
    "Newest_en.properties": "",
  },
  "/path/to/themes/newest-theme/src/sass": Object {
    "_partials.scss": "// Dependency styles
@import \\"?smart-import-partials\\";

// Own partials
",
    "_variables.scss": "// Own variable imports first

// Dependency variables
@import \\"?smart-import-variables\\";
",
    "newest.scss": "/*! Theme newest */
// ### VARIABLES ###

// Own variables (need to be loaded first, so default values can be overridden)
// @see http://sass-lang.com/documentation/file.SASS_REFERENCE.html#Variable_Defaults___default

@import \\"variables\\";

// ### PARTIALS ###

@import \\"partials\\";
",
    "preview.scss": "/*! Theme newest: Preview Styles */
// ### VARIABLES ###

// Own variables (need to be loaded first, so default values can be overridden)
// @see https://sass-lang.com/documentation/file.SASS_REFERENCE.html#Variable_Defaults___default

//@import \\"variables/...\\";

// Dependency variables

@import \\"?smart-import-variables\\";

// ... add third-party dependencies here (after smart-import-variables)

// ### PARTIALS ###

// Dependency partials

// ... add third-party dependencies here (before smart-import-partials)

@import \\"?smart-import-partials\\";

// Own partials

//@import \\"partials/...\\";
",
  },
}
`;

exports[`createTheme() with parent theme creates folder structure and files of a new theme 1`] = `
Array [
  "/path/to/themes/newest-theme/src/js",
  "/path/to/themes/newest-theme/src/sass",
  "/path/to/themes/newest-theme/src/sass/partials",
  "/path/to/themes/newest-theme/src/sass/variables",
  "/path/to/themes/newest-theme/src/img",
  "/path/to/themes/newest-theme/src/fonts",
  "/path/to/themes/newest-theme/src/l10n",
  "/path/to/themes/newest-theme/src/settings",
  "/path/to/themes/newest-theme/src/templates/com.coremedia.blueprint.common.contentbeans",
]
`;

exports[`createTheme() with parent theme creates folder structure and files of a new theme 2`] = `
Object {
  "/path/to/themes/newest-theme": Object {
    "package.json": "{
  \\"name\\": \\"@coremedia/newest-theme\\",
  \\"version\\": \\"1.0.0\\",
  \\"private\\": true,
  \\"scripts\\": {
    \\"build\\": \\"cross-env NODE_OPTIONS=--openssl-legacy-provider webpack\\",
    \\"deploy\\": \\"cm theme-importer deploy-theme\\",
    \\"start\\": \\"cm monitor\\",
    \\"prettier\\": \\"prettier --write \\\\\\"src/**/*.{js,json}\\\\\\"\\",
    \\"theme-importer\\": \\"cm theme-importer\\"
  },
  \\"dependencies\\": {
    \\"parent-theme\\": \\"^1.0.0\\",
    \\"@coremedia/cm-cli\\": \\"workspace:2.0.0\\",
    \\"@coremedia/theme-utils\\": \\"workspace:3.0.0\\",
    \\"cross-env\\": \\"^7.0.3\\",
    \\"webpack\\": \\"^4.46.0\\"
  },
  \\"devDependencies\\": {
    \\"prettier\\": \\"^2.4.1\\"
  },
  \\"main\\": \\"src/js/index.js\\",
  \\"coremedia\\": {
    \\"type\\": \\"theme\\",
    \\"init\\": \\"src/js/newest.js\\"
  },
  \\"browserslist\\": [
    \\"last 1 Chrome version\\",
    \\"last 1 Firefox version\\",
    \\"last 1 Edge version\\"
  ]
}",
    "theme.config.json": "{
  \\"name\\": \\"newest\\",
  \\"styles\\": [
    {
      \\"type\\": \\"webpack\\",
      \\"src\\": \\"src/sass/newest.scss\\"
    }
  ],
  \\"scripts\\": [
    {
      \\"type\\": \\"webpack\\",
      \\"src\\": \\"src/js/newest.js\\"
    }
  ],
  \\"l10n\\": {
    \\"bundleEncoding\\": \\"UTF-8\\",
    \\"bundleNames\\": [
      \\"Newest\\",
      \\"Bricks\\"
    ]
  }
}",
    "webpack.config.js": "const webpackConfig = require(\\"parent-theme/webpack.config.js\\");

module.exports = webpackConfig;",
  },
  "/path/to/themes/newest-theme/src/js": Object {
    "index.js": "/*! Theme newest */
import \\"./newest.js\\";
",
    "newest.js": "//import $ from \\"jquery\\";
//import * as utils from \\"@coremedia/brick-utils\\";

// --- JQUERY DOCUMENT READY -------------------------------------------------------------------------------------------

//$(function () {
//  utils.log(\\"Theme newest is used.\\");
//});
",
    "preview.js": "/*! Theme newest: Preview JS */
// add preview specific code here...
",
  },
  "/path/to/themes/newest-theme/src/l10n": Object {
    "Newest_en.properties": "",
  },
  "/path/to/themes/newest-theme/src/sass": Object {
    "_partials.scss": "// Dependency styles
@import \\"?smart-import-partials\\";

// Own partials
",
    "_variables.scss": "// Own variable imports first

// Dependency variables
@import \\"?smart-import-variables\\";
",
    "newest.scss": "/*! Theme newest */
// ### VARIABLES ###

// Own variables (need to be loaded first, so default values can be overridden)
// @see http://sass-lang.com/documentation/file.SASS_REFERENCE.html#Variable_Defaults___default

@import \\"variables\\";

// ### PARTIALS ###

@import \\"partials\\";
",
    "preview.scss": "/*! Theme newest: Preview Styles */
// ### VARIABLES ###

// Own variables (need to be loaded first, so default values can be overridden)
// @see https://sass-lang.com/documentation/file.SASS_REFERENCE.html#Variable_Defaults___default

//@import \\"variables/...\\";

// Dependency variables

@import \\"?smart-import-variables\\";

// ... add third-party dependencies here (after smart-import-variables)

// ### PARTIALS ###

// Dependency partials

// ... add third-party dependencies here (before smart-import-partials)

@import \\"?smart-import-partials\\";

// Own partials

//@import \\"partials/...\\";
",
  },
}
`;

exports[`createThemeFiles() creates files of a new theme including bricks 1`] = `
Object {
  "/path/to/themes/newest-theme": Object {
    "package.json": "{
  \\"name\\": \\"@coremedia/newest-theme\\",
  \\"version\\": \\"1.0.0\\",
  \\"private\\": true,
  \\"scripts\\": {
    \\"build\\": \\"cross-env NODE_OPTIONS=--openssl-legacy-provider webpack\\",
    \\"deploy\\": \\"cm theme-importer deploy-theme\\",
    \\"start\\": \\"cm monitor\\",
    \\"prettier\\": \\"prettier --write \\\\\\"src/**/*.{js,json}\\\\\\"\\",
    \\"theme-importer\\": \\"cm theme-importer\\"
  },
  \\"__comment__dependencies__\\": {
    \\"__comment__\\": \\"List of all brick dependencies. In order to add a dependency, move the entry to 'dependencies'\\",
    \\"some-brick\\": \\"^1.0.0\\",
    \\"@coremedia/brick-utils\\": \\"^1.0.0\\",
    \\"jquery\\": \\"^3.6.0\\"
  },
  \\"dependencies\\": {
    \\"active-brick\\": \\"^1.0.0\\",
    \\"@coremedia/cm-cli\\": \\"workspace:2.0.0\\",
    \\"@coremedia/theme-utils\\": \\"workspace:3.0.0\\",
    \\"cross-env\\": \\"^7.0.3\\",
    \\"webpack\\": \\"^4.46.0\\"
  },
  \\"devDependencies\\": {
    \\"prettier\\": \\"^2.4.1\\"
  },
  \\"main\\": \\"src/js/index.js\\",
  \\"coremedia\\": {
    \\"type\\": \\"theme\\",
    \\"init\\": \\"src/js/newest.js\\"
  },
  \\"browserslist\\": [
    \\"last 1 Chrome version\\",
    \\"last 1 Firefox version\\",
    \\"last 1 Edge version\\"
  ]
}",
    "theme.config.json": "{
  \\"name\\": \\"newest\\",
  \\"styles\\": [
    {
      \\"type\\": \\"webpack\\",
      \\"src\\": \\"src/sass/newest.scss\\"
    }
  ],
  \\"scripts\\": [
    {
      \\"type\\": \\"webpack\\",
      \\"src\\": \\"src/js/newest.js\\"
    }
  ],
  \\"l10n\\": {
    \\"bundleEncoding\\": \\"UTF-8\\",
    \\"bundleNames\\": [
      \\"Newest\\",
      \\"Bricks\\"
    ]
  }
}",
    "webpack.config.js": "const { webpackConfig } = require(\\"@coremedia/theme-utils\\");

module.exports = webpackConfig;",
  },
  "/path/to/themes/newest-theme/src/js": Object {
    "index.js": "/*! Theme newest */
import \\"./newest.js\\";
",
    "newest.js": "//import $ from \\"jquery\\";
//import * as utils from \\"@coremedia/brick-utils\\";

// --- JQUERY DOCUMENT READY -------------------------------------------------------------------------------------------

//$(function () {
//  utils.log(\\"Theme newest is used.\\");
//});
",
    "preview.js": "/*! Theme newest: Preview JS */
// add preview specific code here...
",
  },
  "/path/to/themes/newest-theme/src/l10n": Object {
    "Newest_en.properties": "",
  },
  "/path/to/themes/newest-theme/src/sass": Object {
    "_partials.scss": "// Dependency styles
@import \\"?smart-import-partials\\";

// Own partials
",
    "_variables.scss": "// Own variable imports first

// Dependency variables
@import \\"?smart-import-variables\\";
",
    "newest.scss": "/*! Theme newest */
// ### VARIABLES ###

// Own variables (need to be loaded first, so default values can be overridden)
// @see http://sass-lang.com/documentation/file.SASS_REFERENCE.html#Variable_Defaults___default

@import \\"variables\\";

// ### PARTIALS ###

@import \\"partials\\";
",
    "preview.scss": "/*! Theme newest: Preview Styles */
// ### VARIABLES ###

// Own variables (need to be loaded first, so default values can be overridden)
// @see https://sass-lang.com/documentation/file.SASS_REFERENCE.html#Variable_Defaults___default

//@import \\"variables/...\\";

// Dependency variables

@import \\"?smart-import-variables\\";

// ... add third-party dependencies here (after smart-import-variables)

// ### PARTIALS ###

// Dependency partials

// ... add third-party dependencies here (before smart-import-partials)

@import \\"?smart-import-partials\\";

// Own partials

//@import \\"partials/...\\";
",
  },
}
`;

exports[`createThemeFolderStructure() creates folder structure of a new theme 1`] = `
Array [
  "/path/to/themes/newest-theme/src/js",
  "/path/to/themes/newest-theme/src/sass",
  "/path/to/themes/newest-theme/src/sass/partials",
  "/path/to/themes/newest-theme/src/sass/variables",
  "/path/to/themes/newest-theme/src/img",
  "/path/to/themes/newest-theme/src/fonts",
  "/path/to/themes/newest-theme/src/l10n",
  "/path/to/themes/newest-theme/src/settings",
  "/path/to/themes/newest-theme/src/templates/com.coremedia.blueprint.common.contentbeans",
]
`;
